src/tools/bin/pic2xdmf.py

```diff
@@ -151,8 +151,8 @@ def join_references_from_components(node_list, original_nodes_map, prefix,
             extra_grid = "Grid[@Name='{}']/".format(
                 parent_node.getAttribute("Name"))
 
-        node_text = "/Xdmf/Domain/Grid/{}Attribute[@Name='{}']/" + \
-                    "DataItem[1]".format(extra_grid, orig_name)
+        node_text = ("/Xdmf/Domain/Grid/{}Attribute[@Name='{}']/"
+                     "DataItem[1]").format(extra_grid, orig_name)
```

Comment:
OPTIONAL:
A plus sign `+` is optional here but would improve readability of the code to emphasize that here the second string is attached.

Reply:
same below

Reply:
I would not add it since it is superfluous like additional brackets.
the way it is now is the common way to do it properly.

Reply:
Okay